### PR TITLE
Meta: bump ecmarkup to 15.0.0 and apply new formatting/lint rules

### PR DESCRIPTION
**Pull request by bakkot**
Depends on https://github.com/tc39/ecma262/pull/2896, and is based on that branch.

I recommend reviewing commits individually, and commits should land individually.

The "Markup: replace HTML entities" commit was generated by running `npm run format`; no manual changes were applied. Sidebar: we sure do have a lot of fancy quote marks.

The "Markup: mark certain non-obvious variable declarations" commit is necessary to get the linter's new "variables must be defined" lint to pass; the ten lines it touches make use of variables in non-obvious ways that I didn't want to teach the linter about.

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -29911,7 +29911,7 @@ <h1>Function ( _p1_, _p2_, &hellip; , _pn_, _body_ )</h1>
         <p>This function performs the following steps when called:</p>
         <emu-alg>
           1. Let _C_ be the active function object.
-          1. Let _args_ be the _argumentsList_ that was passed to this function by [[Call]] or [[Construct]].
+          1. [declared="argumentsList"] Let _args_ be the _argumentsList_ that was passed to this function by [[Call]] or [[Construct]].
```
Comment (michaelficarra) [inline spec.html:29914]: FYI see https://github.com/tc39/ecma262/pull/2902

File: spec.html [hunk 2]
```diff
@@ -2356,7 +2356,7 @@ <h1>
           <emu-alg>
             1. If _d_ is *0*<sub>ℤ</sub>, throw a *RangeError* exception.
             1. If _n_ is *0*<sub>ℤ</sub>, return *0*<sub>ℤ</sub>.
-            1. Let _r_ be the BigInt defined by the mathematical relation _r_ = _n_ - (_d_ &times; _q_) where _q_ is a BigInt that is negative only if _n_/_d_ is negative and positive only if _n_/_d_ is positive, and whose magnitude is as large as possible without exceeding the magnitude of the true mathematical quotient of _n_ and _d_.
+            1. [declared="q"] Let _r_ be the BigInt defined by the mathematical relation _r_ = _n_ - (_d_ &times; _q_) where _q_ is a BigInt that is negative only if _n_/_d_ is negative and positive only if _n_/_d_ is positive, and whose magnitude is as large as possible without exceeding the magnitude of the true mathematical quotient of _n_ and _d_.
```
Comment (michaelficarra) [inline spec.html:2359]:
```suggestion
            1. Let _r_ and _q_ be defined by the mathematical relation _r_ = _n_ - (_d_ &times; _q_) where _q_ is a BigInt that is negative only if _n_/_d_ is negative and positive only if _n_/_d_ is positive, and whose magnitude is as large as possible without exceeding the magnitude of the true mathematical quotient of _n_ and _d_.
```

as above

File: spec.html [hunk 3]
```diff
@@ -1932,7 +1932,7 @@ <h1>
             1. If _d_ is *+0*<sub>𝔽</sub> or _d_ is *-0*<sub>𝔽</sub>, return *NaN*.
             1. If _n_ is *+0*<sub>𝔽</sub> or _n_ is *-0*<sub>𝔽</sub>, return _n_.
             1. Assert: _n_ and _d_ are finite and non-zero.
-            1. Let _r_ be ℝ(_n_) - (ℝ(_d_) &times; _q_) where _q_ is an integer that is negative if and only if _n_ and _d_ have opposite sign, and whose magnitude is as large as possible without exceeding the magnitude of ℝ(_n_) / ℝ(_d_).
+            1. [declared="q"] Let _r_ be ℝ(_n_) - (ℝ(_d_) &times; _q_) where _q_ is an integer that is negative if and only if _n_ and _d_ have opposite sign, and whose magnitude is as large as possible without exceeding the magnitude of ℝ(_n_) / ℝ(_d_).
```
[the remark this answers is not in the document]
Comment (bakkot) [reply spec.html:1935]: I wouldn't want our choice of wording to be motivated by what's easiest for tooling. If you think that wording would be better for human readers, not just tools, feel free to open a PR with that change, but I consider it out of scope for this PR.

[the remark this answers is not in the document]
Comment (michaelficarra) [reply spec.html:1935]: I do think it would be better for a human reader. See #2911.

**Review by michaelficarra — APPROVED** [review]
LGTM but I'd like to see #2902 and #2911 resolved first.

**Comment by bakkot** [comment]
This now incorporates changes necessary to make https://github.com/tc39/ecma262/pull/2902 pass. It also adds a new marked non-obvious declaration for an alias introduced in https://github.com/tc39/ecma262/pull/2781 (in a fixup commit, which should get squashed into the "mark certain non-obvious variable declarations" commit when landing).

It's ready to go as soon as https://github.com/tc39/ecma262/pull/2896 lands, and will then un-block https://github.com/tc39/ecma262/pull/2902.

**Pull request merged**